pkg/events/derive/trigger.go

```diff
@@ -0,0 +1,55 @@
+package derive
+
+import (
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/counter"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/types/trace"
+	"sync"
+)
+
+// Context is a struct used to store the triggering events' context
+type context struct {
+	Map     map[uint64]trace.Event
+	Mutex   sync.RWMutex
+	Counter counter.Counter
+}
+
+var invokedContext = context{make(map[uint64]trace.Event), sync.RWMutex{}, counter.Counter(1)}
+
+func StoreEventContext(event trace.Event) uint64 {
+	// Initial event - no need to save
+	if event.Timestamp == 0 {
+		return 0
+	}
+	contextMapID := uint64(invokedContext.Counter.Read())
+	invokedContext.Counter.Increment(1)
+	invokedContext.Mutex.Lock()
+	invokedContext.Map[contextMapID] = event
+	invokedContext.Mutex.Unlock()
+	return contextMapID
+}
+
+func GetEventContext(contextID uint64) (trace.Event, error) {
+	invokedContext.Mutex.RLock()
+	contextEvent, ok := invokedContext.Map[contextID]
+	invokedContext.Mutex.RUnlock()
+	if !ok {
+		return trace.Event{}, fmt.Errorf("caller_context_id arg not in context map")
+	}
+	// Remove from map to avoid memory leak
+	delete(invokedContext.Map, contextID)
+	return contextEvent, nil
+}
+
+func withInvokingContext(event *trace.Event) (trace.Event, error) {
```

Comment:
This function is targeting a specific event (hooked syscall) and, yet, it is implemented like a generic function to the package. Should we do a switch case here ?

pseudo:
```
switch (event.ID) {
case hookedsyscall:
do what you did
default:
return original context (or error: you should not call this)
```

Reply:
This is basically the current logic for defining an invoked event, which other events could be as well, so leaving it generic is the right call I think. 
With that said, the mechanism should be moved, simplifying the derivation package would go naturally with that.

Reply:
It's a generic function because it's used both in `hooked_syscalls` and in `hooked_seq_ops`.
Also I think it's best that we keep it generic for future usages (in any way we need to pass this to `singleEventDeriveFunc` so it's the developers' choice if to use the original or this.